ninja-maven-plugin/src/main/java/ninja/maven/NinjaRunMojo.java

```diff
@@ -297,20 +358,31 @@ else if (file.getName().endsWith(".jar") && this.watchAllClassPathJars) {
         
         jvmArguments.add(systemPropertyDevMode);
 
-        String portSelection
-                = "-D" + NinjaJetty.COMMAND_LINE_PARAMETER_NINJA_PORT + "=" + port;
-
-        jvmArguments.add(portSelection);
+        if (port != null) {
+            String portSelection
+                    = "-D" + NinjaJetty.KEY_NINJA_PORT + "=" + port;
+            jvmArguments.add(portSelection);
+        }
         
-        if (contextPath != null) {
-            String systemPropertyContextPath = "-Dninja.context=" + contextPath;
+        if (getContextPath() != null) {
+            String systemPropertyContextPath = "-Dninja.context=" + getContextPath();
             jvmArguments.add(systemPropertyContextPath);
         }
         
+        if (jvmArgs != null) {
+            // use excellent DrJava library to tokenize arguments (do not keep tokens)
+            List<String> tokenizedArgs = ArgumentTokenizer.tokenize(jvmArgs, false);
```

Comment:
Tokenizing shell-like command lines is actually not a trivial problem.  You need to deal with spaces, commas, arguments inside quotes, etc.  Everything on the Internet pointed to DrJava's implementation as the only one not broken.  Looks like even Apache Commons is notorious for being wrong.  It's unfortunately not distributed on Maven central, so copying and pasting it seemed like the best solution.  It's BSD licensed (which is compatible w/ Apache) and other than the package, it wasn't modified.